src/jsm.c

```diff
@@ -3602,6 +3628,60 @@ js_PauseConsumer(jsConsumerPauseResponse **new_cpr, jsCtx *js,
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+natsStatus
+js_UnpinConsumer(jsCtx *js, const char *stream, const char *consumer, const char *group,
+                  jsOptions *opts, jsErrCode *errCode)
+{
+    natsStatus          s       = NATS_OK;
+    char                *subj   = NULL;
+    bool                freePfx = false;
+    natsConnection      *nc     = NULL;
+    natsMsg             *resp   = NULL;
+    bool                success = false;
+    jsOptions           o;
+    char                jsonBuf[64];
+
+    if (errCode != NULL)
+        *errCode = 0;
+
+    if (js == NULL)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    s = _checkStreamName(stream);
+    IFOK(s, js_checkConsName(consumer, false));
+    IFOK(s, nats_validateLimitedTerm("group", group));
+    if (s != NATS_OK)
+        return NATS_UPDATE_ERR_STACK(s);
+
+    s = js_setOpts(&nc, &freePfx, js, opts, &o);
+    if (s == NATS_OK)
+    {
+        if (nats_asprintf(&subj, jsApiConsumerUnpinT,
+                          js_lenWithoutTrailingDot(o.Prefix), o.Prefix,
+                          stream, consumer) < 0 )
+        {
+            s = nats_setDefaultError(NATS_NO_MEMORY);
+        }
+        if (freePfx)
+            NATS_FREE((char*) o.Prefix);
+    }
+
+    snprintf(jsonBuf, sizeof(jsonBuf), "{\"group\":\"%s\"}", group);
```

Comment:
Should be only if `if (s == NATS_OK)` because you could have set it to "NATS_NO_MEMORY" when creating the subject above.

Reply:
Also, adding a little comment that `group` has already been verified to be no more than 16 characters, so `jsonBuf` set to 64 is big enough.